modules/infinite-scroll/infinity.js

```diff
@@ -170,24 +244,71 @@
 	 */
 	Scroller.prototype.thefooter = function() {
 		var self = this,
-			width;
+			pageWrapper,
+			footerContainer,
+			width,
+			sourceBottom,
+			targetBottom;
 
 		// Check if we have an id for the page wrapper
-		if ( $.type( this.footer.wrap ) === 'string' ) {
-			width = $( 'body #' + this.footer.wrap ).outerWidth( false );
+		if ( 'string' === typeof this.footer.wrap ) {
+			try {
+				pageWrapper = document.getElementById( this.footer.wrap );
+				width = pageWrapper.getBoundingClientRect();
+				width = width.width;
+			} catch ( err ) {
+				width = 0;
+			}
 
 			// Make the footer match the width of the page
 			if ( width > 479 ) {
-				this.footer.find( '.container' ).css( 'width', width );
+				footerContainer = this.footer.el.querySelector( '.container' );
+				if ( footerContainer ) {
+					footerContainer.style.width = width + 'px';
+				}
 			}
 		}
 
 		// Reveal footer
-		if ( this.window.scrollTop() >= 350 ) {
-			self.footer.animate( { bottom: 0 }, 'fast' );
-		} else if ( this.window.scrollTop() < 350 ) {
-			self.footer.animate( { bottom: '-50px' }, 'fast' );
+		sourceBottom = parseInt( self.footer.el.style.bottom || -50, 10 );
+		targetBottom = this.window.pageYOffset >= 350 ? 0 : -50;
+
+		if ( sourceBottom !== targetBottom ) {
+			self.animate( function( progress ) {
+				var currentBottom = sourceBottom + ( targetBottom - sourceBottom ) * progress;
+				self.footer.el.style.bottom = currentBottom + 'px';
+
+				if ( 1 === progress ) {
+					sourceBottom = targetBottom;
+				}
+			}, 200 );
+		}
+	};
+
+	/**
+	 * Recursively convert a JS object into URL encoded data.
+	 */
+	Scroller.prototype.urlEncodeJSON = function( obj, prefix ) {
```

Comment:
**Note:** [Usually](https://stackoverflow.com/a/39519299/52982) we could just `POST` with `Content-Type: application/json`, but that would [send the JSON string in the body of the request](https://stackoverflow.com/a/8605464/52982) and we would need to rewrite much of `infinity.php` to adapt to that.

Instead we URI encode the data object, so that we can send the parameters as `application/x-www-form-urlencoded`, which is correctly and automatically decoded by PHP and the `$_POST` (`$_REQUEST`) superglobals are correctly populated.